src/rust/e2e-tests/src/test_utils/context.rs

```diff
@@ -109,45 +199,39 @@ impl E2eTestContext {
     ) -> eyre::Result<SetupResult> {
         tracing::info!(">> Generator Setup for {}", options.test_name);
 
-        let tenant_id = Uuid::new_v4();
+        let tenant_id = self.create_tenant().await?;
 
         // Register an Event Source
-        let event_source = self
-            .event_source_client
-            .create_event_source(CreateEventSourceRequest {
-                display_name: options.test_name.clone(),
-                description: "arbitrary".to_string(),
+        let event_source_id = self
+            .create_event_source(
                 tenant_id,
-            })
+                options.test_name.clone(),
+                "arbitrary".to_string(),
+            )
             .await?;
 
         // Deploy a Generator Plugin that responds to that event_source_id
-        let plugin = {
-            let plugin = self
-                .plugin_registry_client
-                .create_plugin(
-                    PluginMetadata::new(
-                        tenant_id,
-                        options.test_name.clone(),
-                        PluginType::Generator,
-                        Some(event_source.event_source_id),
-                    ),
-                    futures::stream::once(async move { options.generator_artifact }),
+        let generator_id = {
+            let generator_id = self
+                .create_generator(
+                    tenant_id,
+                    options.test_name.clone(),
+                    event_source_id,
+                    options.generator_artifact,
                 )
                 .await?;
 
             if options.should_deploy_generator {
-                self.plugin_registry_client
-                    .deploy_plugin(DeployPluginRequest::new(plugin.plugin_id()))
-                    .await?;
+                self.deploy_generator(generator_id).await?;
             }
-            plugin
+
+            generator_id
         };
 
         let setup_result = SetupResult {
```

Comment:
this is my naming mistake, not yours, but potentially rename to SetupGeneratorResult

Reply:
I tried to avoid having these changes impact a bunch of other services